analyzers/src/SonarAnalyzer.CSharp/Rules/BlazorQueryParameterRoutableComponent.cs

```diff
@@ -0,0 +1,109 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class BlazorQueryParameterRoutableComponent : SonarDiagnosticAnalyzer
+{
+    private const string NoRouteQueryDiagnosticId = "S6803";
+    private const string NoRouteQueryMessageFormat = "Component parameters can only receive query parameter values in routable components.";
+
+    private const string QueryTypeDiagnosticId = "S6797";
+    private const string QueryTypeMessageFormat = "Query parameter type '{0}' is not supported.";
+
+    private static readonly DiagnosticDescriptor S6803Rule = DescriptorFactory.Create(NoRouteQueryDiagnosticId, NoRouteQueryMessageFormat);
+    private static readonly DiagnosticDescriptor S6797Rule = DescriptorFactory.Create(QueryTypeDiagnosticId, QueryTypeMessageFormat);
+
+    private static readonly ISet<KnownType> SupportedQueryTypes = new HashSet<KnownType>
+    {
+        KnownType.System_Boolean,
+        KnownType.System_DateTime,
+        KnownType.System_Decimal,
+        KnownType.System_Double,
+        KnownType.System_Single,
+        KnownType.System_Int32,
+        KnownType.System_Int64,
+        KnownType.System_String,
+        KnownType.System_Guid
+    };
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(S6803Rule, S6797Rule);
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterCompilationStartAction(c =>
+        {
+            if (c.Compilation.GetTypeByMetadataName(KnownType.Microsoft_AspNetCore_Components_RouteAttribute) is not null)
+            {
+                context.RegisterSymbolAction(CheckQueryProperties, SymbolKind.Property);
+            }
+        });
+
+    private static void CheckQueryProperties(SonarSymbolReportingContext c)
+    {
+        var property = (IPropertySymbol)c.Symbol;
+        if (HasComponentParameterAttributes(property))
+        {
+            if (!property.ContainingType.HasAttribute(KnownType.Microsoft_AspNetCore_Components_RouteAttribute))
+            {
+                foreach (var location in property.Locations)
+                {
+                    c.ReportIssue(Diagnostic.Create(S6803Rule, location));
+                }
+            }
+            else if (IsPropertyTypeMismatch(property))
+            {
+                foreach (var propertyType in property.DeclaringSyntaxReferences.Select(r => ((PropertyDeclarationSyntax)r.GetSyntax()).Type))
+                {
+                    c.ReportIssue(Diagnostic.Create(S6797Rule, propertyType.GetLocation(), GetTypeName(propertyType)));
+                }
+            }
+        }
+    }
+
+    private static bool HasComponentParameterAttributes(IPropertySymbol property) =>
```

Comment:
This method is inlined in the most recent implementation of S6803 to address [this comment](https://github.com/SonarSource/sonar-dotnet/pull/8340#discussion_r1389623939).